File: Contribute/Plugin Samples/Plugin Samples/C#/QuickNote/QuickNotePlugin/MainForm.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Xml.Linq;
using SS.Ynote.Classic;
using SS.Ynote.Engine.Framework.Plugins;
using SS.Ynote.Engine.Framework.Plugins.Controls;
using SS.Ynote.Engine.Framework.Plugins.Interface;
using WeifenLuo.WinFormsUI.Docking;

namespace QuickNotePlugin
{
    ///<summary>
    ///Created with Ynote Classic
    ///Compiled with csc.exe
    ///</summary>
    public partial class MainForm : DockContent, IFormPlugin
    {
        #region Constructor

        public MainForm()
        {
            InitializeComponent();
            this.newToolStripMenuItem1.Click += new EventHandler(menuItem2_Click);
            this.openToolStripMenuItem1.Click += new EventHandler(menuItem3_Click);
            this.importFromMainEditorToolStripMenuItem.Click += new EventHandler(menuItem18_Click);
            this.saveAsToolStripMenuItem1.Click += new EventHandler(menuItem5_Click);
            this.clearAllToolStripMenuItem.Click += new EventHandler(menuItem2_Click);
            this.cutToolStripMenuItem.Click += new EventHandler(menuItem8_Click);
            this.copyToolStripMenuItem.Click += new EventHandler(menuItem9_Click);
            this.pasteToolStripMenuItem.Click += new EventHandler(menuItem10_Click);
            this.selectAllToolStripMenuItem.Click += new EventHandler(menuItem11_Click);
            this.undoToolStripMenuItem.Click += new EventHandler(menuItem7_Click);
            this.fontToolStripMenuItem.Click += new EventHandler(menuItem14_Click);
            this.backColorToolStripMenuItem.Click += new EventHandler(menuItem15_Click);
        }
        #endregion

        #region IFormPlugin Members

        public DockContent Content
        {
            get { return this; }
        }

        public ShowAs ShowAs
        {
            get { return ShowAs.Normal; }
        }


        #endregion

        #region IPlugin Members

        public string Title
        {
            get { return "QuickNotePlugin"; }
        }
        public string Description
        {
            get { return "This is  a Quick Note Plugin for Ynote Classic"; }
        }
        public string Group
        {
            get { return "Plugins"; }
        }
        public string SubGroup
        {
            get { return "Note"; }
        }

        private XElement configuration = new XElement("ThisFormConfig");
        public XElement Configuration
        {
            get { return configuration; }
            set { configuration = value; }
        }

        public string Icon
        {
            //get { return "C:\\Icons\\Folder.ico"; }
            get { return string.Empty; }
        }
        #endregion

        #region Events

        private void menuItem7_Click(object sender, EventArgs e)
        {
            this.txtmain.Undo();
        }

        private void menuItem18_Click(object sender, EventArgs e)
        {
            this.txtmain.Text = Main.UserText;
        }

        private void menuItem2_Click(object sender, EventArgs e)
        {
            this.txtmain.Clear();
        }

        private void menuItem3_Click(object sender, EventArgs e)
        {
            System.Windows.Forms.OpenFileDialog o = new System.Windows.Forms.OpenFileDialog();
            o.Filter = "Text Files (*.txt)|*.txt|All Files(*.*)|*.*";
            o.ShowDialog();
            txtmain.Text = File.ReadAllText(o.FileName, Encoding.Default);

        }

        private void menuItem8_Click(object sender, EventArgs e)
        {
            this.txtmain.Cut();
        }

        private void menuItem9_Click(object sender, EventArgs e)
        {
            this.txtmain.Copy();
        }

        private void menuItem10_Click(object sender, EventArgs e)
        {
            this.txtmain.Paste();
        }

        private void menuItem11_Click(object sender, EventArgs e)
        {
            this.txtmain.SelectAll();
        }
        private void menuItem14_Click(object sender, EventArgs e)
        {
            System.Windows.Forms.FontDialog f = new System.Windows.Forms.FontDialog();
            f.ShowEffects = true;
            f.ShowColor = true;
            f.ShowDialog();
            txtmain.Font = f.Font;
            txtmain.ForeColor = f.Color;
  
        }

        private void menuItem15_Click(object sender, EventArgs e)
        {
            System.Windows.Forms.ColorDialog c = new System.Windows.Forms.ColorDialog();
            c.ShowDialog();
            this.txtmain.BackColor = c.Color;
        }

        private void menuItem5_Click(object sender, EventArgs e)
        {
            System.Windows.Forms.SaveFileDialog s = new System.Windows.Forms.SaveFileDialog();
            s.Filter = "All Files (*.*)|*.*";
            s.ShowDialog();
            if (s.FileName != "")
                File.WriteAllText(s.FileName, txtmain.Text, Encoding.Default);
        }

        private void menuItem19_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void exitToolStripMenuItem1_Click(object sender, EventArgs e)
        {
            Application.Exit();
        }
        #endregion
    }
}

File: Contribute/Plugin Samples/VB.NET/QuickNote/MainForm.vb
Imports System.IO
Imports System.ComponentModel
Imports System.Text
Imports SS.Ynote.Classic
Imports SS.Ynote.Engine.Framework.Plugins
Imports SS.Ynote.Engine.Framework.Plugins.Controls
Imports SS.Ynote.Engine.Framework.Plugins.Interface
Imports WeifenLuo.WinFormsUI.Docking

Namespace QuickNotePlugin
    Partial Public Class MainForm
        Inherits DockContent
        Implements IFormPlugin

        #Region "Constructor"

        Public Sub New()
            InitializeComponent()
            AddHandler newToolStripMenuItem1.Click, AddressOf menuItem2_Click
            AddHandler openToolStripMenuItem1.Click, AddressOf menuItem3_Click
            AddHandler importFromMainEditorToolStripMenuItem.Click, AddressOf menuItem18_Click
            AddHandler saveAsToolStripMenuItem1.Click, AddressOf menuItem5_Click
            AddHandler clearAllToolStripMenuItem.Click, AddressOf menuItem2_Click
            AddHandler cutToolStripMenuItem.Click, AddressOf menuItem8_Click
            AddHandler copyToolStripMenuItem.Click, AddressOf menuItem9_Click
            AddHandler pasteToolStripMenuItem.Click, AddressOf menuItem10_Click
            AddHandler selectAllToolStripMenuItem.Click, AddressOf menuItem11_Click
            AddHandler undoToolStripMenuItem.Click, AddressOf menuItem7_Click
            AddHandler fontToolStripMenuItem.Click, AddressOf menuItem14_Click
            AddHandler backColorToolStripMenuItem.Click, AddressOf menuItem15_Click
        End Sub

        #End Region

        #Region "IFormPlugin Members"

        Public ReadOnly Property Content() As DockContent Implements IFormPlugin.Content
            Get
                Return Me
            End Get
        End Property

        Public ReadOnly Property ShowAs() As ShowAs Implements IFormPlugin.ShowAs
            Get
                Return ShowAs.Normal
            End Get
        End Property


        #End Region

        #Region "IPlugin Members"

        Public ReadOnly Property Title() As String Implements SS.Ynote.Engine.Framework.Plugins.Interface.IPlugin.Title
            Get
                Return "QuickNotePlugin"
            End Get
        End Property
        Public ReadOnly Property Description() As String Implements SS.Ynote.Engine.Framework.Plugins.Interface.IPlugin.Description
            Get
                Return "This is  a Quick Note Plugin for Ynote Classic"
            End Get
        End Property
        Public ReadOnly Property Group() As String Implements SS.Ynote.Engine.Framework.Plugins.Interface.IPlugin.Group
            Get
                Return "Plugins"
            End Get
        End Property
        Public ReadOnly Property SubGroup() As String Implements SS.Ynote.Engine.Framework.Plugins.Interface.IPlugin.SubGroup
            Get
                Return "Note"
            End Get
        End Property

        Private configuration_Renamed As New XElement("QuickNote")
        Public Property Configuration() As XElement Implements SS.Ynote.Engine.Framework.Plugins.Interface.IPlugin.Configuration
            Get
                Return configuration_Renamed
            End Get
            Set(ByVal value As XElement)
                configuration_Renamed = value
            End Set
        End Property

        Public ReadOnly Overloads Property Icon() As String Implements SS.Ynote.Engine.Framework.Plugins.Interface.IPlugin.Icon
            'get { return "C:\\Icons\\Folder.ico"; }
            Get
                Return String.Empty
            End Get
        End Property
        #End Region

        #Region "Events"

        Private Sub menuItem7_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.txtmain.Undo()
        End Sub

        Private Sub menuItem18_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.txtmain.Text = Main.ActiveEditor.codebox.Text
        End Sub

        Private Sub menuItem2_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.txtmain.Clear()
        End Sub

        Private Sub menuItem3_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim o As New System.Windows.Forms.OpenFileDialog()
            o.Filter = "Text Files (*.txt)|*.txt|All Files(*.*)|*.*"
            o.ShowDialog()
            txtmain.Text = File.ReadAllText(o.FileName, Encoding.Default)

        End Sub

        Private Sub menuItem8_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.txtmain.Cut()
        End Sub

        Private Sub menuItem9_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.txtmain.Copy()
        End Sub

        Private Sub menuItem10_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.txtmain.Paste()
        End Sub

        Private Sub menuItem11_Click(ByVal sender As Object, ByVal e As EventArgs)
            Me.txtmain.SelectAll()
        End Sub
        Private Sub menuItem14_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim f As New System.Windows.Forms.FontDialog()
            f.ShowEffects = True
            f.ShowColor = True
            f.ShowDialog()
            txtmain.Font = f.Font
            txtmain.ForeColor = f.Color

        End Sub

        Private Sub menuItem15_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim c As New System.Windows.Forms.ColorDialog()
            c.ShowDialog()
            Me.txtmain.BackColor = c.Color
        End Sub

        Private Sub menuItem5_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim s As New System.Windows.Forms.SaveFileDialog()
            s.Filter = "All Files (*.*)|*.*"
            s.ShowDialog()
            If s.FileName <> "" Then
                File.WriteAllText(s.FileName, txtmain.Text, Encoding.Default)
            End If
        End Sub

        Private Sub menuItem19_Click(ByVal sender As Object, ByVal e As EventArgs)
            Close()
        End Sub

        Private Sub exitToolStripMenuItem1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles exitToolStripMenuItem1.Click
            Application.Exit()
        End Sub
        #End Region
    End Class
End Namespace
